### DISCUSSION TITLE
X1nput v3.1

### DISCUSSION THREAD

**Discussion by araghon007** [araghon007/X1nput#28] [General]
- Removed the need for a separate 32 and 64-bit executable
- Added automatic controller detection, enabled by default
- Added support for multiple controllers
- Rearranged the UI
- Fixed some crashes

<hr /><em>This discussion was created from the release <a href='https://github.com/araghon007/X1nput/releases/tag/v3.1'>X1nput v3.1</a>.</em>

**Comment by Kefurant** [comment]
Dear author, hello! I hope you can keep updating and fixing this patch, because it feels completely different from the MS version at the moment, it just syncs the vibration of the handle, not the real trigger vibration

**Reply by araghon007** [reply]
That's because you can't magically add trigger vibrations to a game that wasn't made to use them. All I can do is take the pre-existing vibrations and apply them to the triggers. It's not perfect, but it's as good as it can get for a general solution. Though people could make custom versions of X1nput on a game-to-game basis.

**Comment by Kefurant** [comment]
please please please!!!

**Comment by TheAlgerian** [comment]
Nice tool.
Makes Project Cars 2 more bearable when you're used to how great the driving Forza feels

**Comment by cobaltness** [comment]
This is nice tool but only if i am also able to use it too,
I can't see any games on list when refreshing, i tried downloading .NET framework lastest version and trying to run in administrator mode.

**Reply by araghon007** [reply]
Have you tried using one of the older versions? https://github.com/araghon007/X1nput/releases/tag/v3.0

**Reply by cobaltness** [reply]
Wow, thanks for a fast reply.
I've tried v3.0 version that you attached on but it doesn't seems to find any games.Also i tried your older versions like;
https://github.com/araghon007/X1nput/releases/tag/v2.0
And do the exact thing on readme file any other suggestions ?

**Reply by cobaltness** [reply]
![image](https://user-images.githubusercontent.com/57595244/152643290-4290071c-5ca0-4c1b-ad6f-a11d946c17ae.png)

**Reply by araghon007** [reply]
I'll try to make a version that saves a log of found applications to a file in a bit, hopefully that'll reveal something

**Reply by cobaltness** [reply]
Maybe adding a game by manually could fix it 😓

**Reply by araghon007** [reply]
I'll try that too

**Reply by araghon007** [reply]
Also, what game were you trying to test this with?

**Reply by cobaltness** [reply]
Red Dead Redemption 2 and online

**Reply by araghon007** [reply]
Alright, this one has a manual injection feature. Just click the button on the top left corner and select the game's exe in there. I would suggest against using it in online games though, since you will most likely get banned, especially in a Rockstar game.
[X1nputConfigurator.zip](https://github.com/araghon007/X1nput/files/8008126/X1nputConfigurator.zip)

**Reply by cobaltness** [reply]
Thank you for doing this in less than a hour but defender see it as a serious threat.What do i do now

**Reply by araghon007** [reply]
That's understandable, since it is an injector. I'm surprised the release version doesn't trigger it. Is it the zip file, or the exe that gets detected?

**Reply by araghon007** [reply]
Just checked with VirusTotal, and the release version does in fact not get detected by defender, while the new one does. Dumb

**Reply by cobaltness** [reply]
Yeah i know right 😂.Almost tried every injector no of them dedected as trojan or any type of virus.This one in exe does

**Reply by araghon007** [reply]
Even after removing the logging, it gets detected, which makes me think Defender just got some definitions update, since I did minimal changes to the rest of the code.

**Reply by araghon007** [reply]
Is there any way for you to bypass the warning? I could potentially just update the source code and you could try building it from source, if you have a way to do so.

**Reply by cobaltness** [reply]
I might try on windows safe mode

**Reply by cobaltness** [reply]
> Is there any way for you to bypass the warning? I could potentially just update the source code and you could try building it from source, if you have a way to do so.

I have tried it and used manual injection to some different games and RDR2 too but manual injection only prompts to open the app , like just starting the program you selected

**Reply by cobaltness** [reply]
So .. does this file you attached works standalone or need other dll files as like on v3.0

**Reply by araghon007** [reply]
You need to replace the file in the release folder, so it has all the DLLs

**Comment by fikritheins** [comment]
I tried to inject this into Dirt Rally 2, but sadly it just exit whenever I choose the dirtrally2, and the trigger rumble is non existent also :(
